test/integration/metrics/metrics_test.go

```diff
@@ -261,6 +263,90 @@ func TestAPIServerMetricsLabels(t *testing.T) {
 	}
 }
 
+func TestAPIServerMetricsLabelsWithAllowList(t *testing.T) {
+
+	allowMetricLabels := []struct {
```

Comment:
This setup almost feels like a table test, so can we just convert it to that (similar to the setup of `TestAPIServerMetricsPods()`)?

Reply:
Updated.